File: 0828-113-1718-construct-the-lexicographically-largest-valid-sequence/0828-113-1718-construct-the-lexicographically-largest-valid-sequence.py
class Solution:
    def constructDistancedSequence(self, n: int) -> List[int]:
        v = [0]*(2*n-1)
        vis = set()
        def solve(idx,n):
            if idx == len(v):
                return True
            if v[idx] != 0:
                return solve(idx+1,n)
            for i in range(n,0,-1):
                if i in vis:
                    continue
                nxt = idx
                if i >1:
                    nxt+=i
                if nxt >= 2*n-1 or v[nxt] != 0:
                    continue
                vis.add(i)
                v[idx] = v[nxt] = i
                if solve(idx+1,n) == True:
                    return True
                vis.remove(i)
                v[idx] = v[nxt] = 0
            return False
       
        solve(0,n)
        return v

File: 0828-113-1718-construct-the-lexicographically-largest-valid-sequence/0828-113-1718-construct-the-lexicographically-largest-valid-sequence.js
var constructDistancedSequence = function(n) {
    let v = new Array(2 * n - 1).fill(0);
    let vis = new Array(n + 1).fill(false);

    function solve(idx) {
        if (idx === 2 * n - 1) return true;
        if (v[idx] !== 0) return solve(idx + 1);

        for (let i = n; i >= 1; i--) {
            if (vis[i]) continue;
            let nxt = i === 1 ? idx : idx + i;
            if (nxt >= v.length || v[nxt] !== 0) continue;

            v[idx] = v[nxt] = i;
            vis[i] = true;

            if (solve(idx + 1)) return true;

            v[idx] = v[nxt] = 0;
            vis[i] = false;
        }
        return false;
    }

    solve(0);
    return v;
};
